cluster/gce/gci/configure-helper-testcase.go

```diff
@@ -0,0 +1,200 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gci
+
+import (
+	"github.com/golang/glog"
+
+	goruntime "runtime"
+
+	"testing"
+
+	"text/template"
+
+	"io/ioutil"
+
+	"k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"os"
+	"path/filepath"
+
+	"fmt"
+	"io"
+	"os/exec"
+	"path"
+
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+)
+
+const (
+	envScriptFileName         = "kube-env"
+	configureHelperScriptName = "configure-helper.sh"
+)
+
+type ManifestTestCase struct {
+	pod                 v1.Pod
+	envScriptPath       string
+	manifest            string
+	auxManifests        []string
+	kubeHome            string
+	manifestSources     string
+	manifestDestination string
+	manifestTemplate    string
+	manifestFuncName    string
+	manifestTemplateDir string
+	t                   *testing.T
+}
+
+func NewManifestTestCase(manifestUnderTest, funcUnderTest string, auxManifests []string, t *testing.T) *ManifestTestCase {
+	c := &ManifestTestCase{}
+	c.t = t
+	d, err := ioutil.TempDir("/tmp", "")
+	if err != nil {
+		c.t.Fatalf("Failed to create temp directory: %v", err)
+	}
+
+	c.kubeHome = d
+	glog.V(3).Infof("KUBE_HOME of the test case is %s", c.kubeHome)
+
+	c.manifestFuncName = funcUnderTest
+	glog.V(3).Infof("configure-helper.sh function under test is: %s", c.manifestFuncName)
+
+	c.manifest = manifestUnderTest
+	glog.V(3).Infof("Manifest under test is %s", manifestUnderTest)
+
+	// Some manifests consume other manifests. For example, kube-apiserver.manifest may consume kms-plugin-container.manifest.
+	c.auxManifests = auxManifests
+	glog.V(3).Infof("Auxiliary manifests: %v", c.auxManifests)
+
+	c.envScriptPath = filepath.Join(c.kubeHome, envScriptFileName)
+	glog.V(3).Infof("kube-env variables will be sourced by bash from %s", c.envScriptPath)
+
+	c.manifestSources = path.Join(filepath.Join(c.kubeHome, "kube-manifests", "kubernetes", "gci-trusty"))
+	glog.V(3).Infof("Function under test will access the manifest from %s", c.manifestSources)
+
+	// full path of this go file
+	_, testFilePath, _, _ := goruntime.Caller(0)
+	c.manifestTemplateDir = filepath.Join(path.Dir(path.Dir(testFilePath)), "manifests")
+	c.manifestTemplate = filepath.Join(c.manifestTemplateDir, c.manifest)
+	glog.V(3).Infof("Test case will use a copy of this manifest: %s", c.manifestTemplate)
+
+	c.manifestDestination = filepath.Join(c.kubeHome, "etc", "kubernetes", "manifests", c.manifest)
+	glog.V(3).Infof("The resulting manifest will be created in %s", c.manifestDestination)
+
+	c.mustCopyFromTemplate()
+	c.mustCopyAuxFromTemplate()
+	c.mustCreateManifestDstDir()
+
+	return c
+}
+
+func (c *ManifestTestCase) mustCopyFromTemplate() {
+	err := os.MkdirAll(c.manifestSources, os.ModePerm)
+	if err != nil {
```

Comment:
Merge these two lines
if err := os.MkdirAll(...); err != nil {
...
}

Same everywhere in this file

Reply:
Done.